lib/webauth.js

```diff
@@ -125,10 +131,60 @@ StatusResponse.prototype = {
   keyField: "authenticated"
 }
 
+const SESSION_ACTION_TYPE_AUTHENTICATE = 1;
+const SESSION_ACTION_TYPE_REFRESH = 2;
+
 /*
  * Assumes req.session is there and behaves as it should
  */
 module.exports = function(authManager) {
+  const _authenticateOrRefresh = Promise.coroutine(function*(req, res, type) {
```

Comment:
I don't see a problem just adding a method called _authenticateOrRefresh () to the  returned object.  I've been following the convention that _foo means it's private.

We'll get compile time checking for private fields/variables after TypeScript migration anyway, so I think the OO style is preferable than nested closures.

Reply:
It's better to do this for defensive programming... typescript doesn't actually make private fields private at runtime, and I worry about non-TS code being abusive.